Content.Server/GameObjects/Components/Mobs/StunnableComponent.cs

```diff
@@ -0,0 +1,157 @@
+using System;
+using System.Threading;
+using Content.Server.GameObjects.EntitySystems;
+using Content.Server.Interfaces.GameObjects;
+using Content.Server.Mobs;
+using Content.Shared.Audio;
+using Content.Shared.GameObjects.Components.Mobs;
+using Robust.Server.GameObjects;
+using Robust.Server.GameObjects.EntitySystems;
+using Robust.Shared.Audio;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Timers;
+using Robust.Shared.IoC;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+using Timer = Robust.Shared.Timers.Timer;
+
+namespace Content.Server.GameObjects.Components.Mobs
+{
+    [RegisterComponent]
+    public class StunnableComponent : Component, IActionBlocker, IAttackHand
+    {
+        [Dependency] private IEntitySystemManager _entitySystemManager;
+        [Dependency] private ITimerManager _timerManager;
+
+        private bool _stunned = false;
+        private bool _knocked = false;
+        private bool _canHelp = true;
+
+        private float _stunCap = 20f;
+        private float _knockdownCap = 20f;
+        private float _helpKnockdownRemove = 1f;
+        private float _helpInterval = 1f;
+
+        private float _stunnedTimer = 0f;
+        private float _knockdownTimer = 0f;
+
+        public override string Name => "Stunnable";
+
+        [ViewVariables] public bool Stunned => _stunned;
+        [ViewVariables] public bool KnockedDown => _knocked;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+            serializer.DataField(ref _stunCap, "stunCap", 20f);
+            serializer.DataField(ref _knockdownCap, "knockdownCap", 20f);
+            serializer.DataField(ref _helpInterval, "helpInterval", 1f);
+            serializer.DataField(ref _helpKnockdownRemove, "helpKnockdownRemove", 1f);
+        }
+
+        public void Stun(float seconds)
+        {
+            seconds = Math.Min(seconds + _stunnedTimer, _stunCap);
+
+            StandingStateHelper.DropAllItemsInHands(Owner);
```

Comment:
I'm a bit surprised by this - do stuns (as in slowdown attacks, not the knockdown version) make you drop stuff in SS13?

Reply:
Problem is, this isn't a "slowdown attack" but a "can't do most actions at all" stun. I asked this in the discord and they told me to do it, but I can remove it if people want me to.

Reply:
Your review inspired me to make a slowdown stun actually-- I'm resolving this